src/main/java/org/fogbowcloud/manager/core/threads/AttendOpenOrdersThread.java

```diff
@@ -0,0 +1,138 @@
+package org.fogbowcloud.manager.core.threads;
+
+import java.util.Properties;
+
+import org.apache.log4j.Logger;
+import org.fogbowcloud.manager.core.constants.ConfigurationConstants;
+import org.fogbowcloud.manager.core.constants.DefaultConfigurationConstants;
+import org.fogbowcloud.manager.core.instanceprovider.InstanceProvider;
+import org.fogbowcloud.manager.core.models.orders.Order;
+import org.fogbowcloud.manager.core.models.orders.OrderRegistry;
+import org.fogbowcloud.manager.core.models.orders.OrderState;
+import org.fogbowcloud.manager.core.models.orders.instances.OrderInstance;
+
+public class AttendOpenOrdersThread extends Thread {
+
+	private InstanceProvider localInstanceProvider;
+	private InstanceProvider remoteInstanceProvider;
+
+	private OrderRegistry orderRegistry;
+
+	private String localMemberId;
+
+	/**
+	 * Attribute that represents the thread sleep time when there is no Orders
+	 * to be processed.
+	 */
+	private Long sleepTime;
+
+	private static final Logger LOGGER = Logger.getLogger(AttendOpenOrdersThread.class);
+
+	public AttendOpenOrdersThread(InstanceProvider localInstanceProvider, InstanceProvider remoteInstanceProvider,
+			OrderRegistry orderRegistry, String localMemberId, Properties properties) {
+		this.localInstanceProvider = localInstanceProvider;
+		this.remoteInstanceProvider = remoteInstanceProvider;
+		this.orderRegistry = orderRegistry;
+		this.localMemberId = localMemberId;
+
+		String schedulerPeriodStr = properties.getProperty(ConfigurationConstants.OPEN_ORDERS_SLEEP_TIME_KEY,
+				DefaultConfigurationConstants.OPEN_ORDERS_SLEEP_TIME);
+		this.sleepTime = Long.valueOf(schedulerPeriodStr);
+	}
+
+	@Override
+	public void run() {
+		while (true) {
+			try {
+				Order order = this.orderRegistry.getNextOrderByState(OrderState.OPEN);
+				if (order != null) {
+					this.processOpenOrder(order);
+				} else {
+					LOGGER.info("There is no Open Order to be processed, sleeping Attend Open Orders Thread...");
+					Thread.sleep(this.sleepTime);
+				}
+			} catch (Exception e) {
+				LOGGER.error("Error while trying to sleep Attend Order Thread", e);
+			}
+		}
+	}
+
+	/**
+	 * Method that try to get an Instance for an Open Order. This method can
+	 * generate a race condition. For example: a user can delete a Open Order
+	 * while this method is trying to get an Instance for this Order.
+	 */
+	protected void processOpenOrder(Order order) {
+		synchronized (order) {
+			OrderState orderState = order.getOrderState();
+
+			if (orderState.equals(OrderState.OPEN)) {
+				LOGGER.info("Trying to get an Instance for Order [" + order.getId() + "]");
+
+				try {
+					InstanceProvider instanceProvider = this.getInstanceProviderForOrder(order);
+
+					LOGGER.info("Processing Order [" + order.getId() + "]");
+					order.processOpenOrder(instanceProvider);
```

Comment:
@paulofelipefeitosa as we discussed earlier. move the procesOpenOrder code from the order to this class.